src/main/java/com/google/api/generator/engine/escaper/CommentEscaper.java

```diff
@@ -32,16 +32,41 @@
             .addEscape('\\', "\\\\")
             .build();
 
-    private SpecialEscaper() {}
+    private SpecialCharEscaper() {}
 
     @Override
     public String escape(String sourceString) {
       return escaper.escape(sourceString);
     }
   }
+  // HtmlEscaper is separately defined from SpecialCharEscaper even if they extend the same Escaper,
+  // because HtmlEscaper would be only called by JavaDocComment.
+  private static class HtmlEscaper extends Escaper {
+    // Based on the observation of the generated java files, we escape the following
+    // five characters by html escaper. We do not directly use guava HtmlEscapers here because
+    // it only escapes`<>&\"'` as specified by HTML 4.01.
+    private static final Escaper escaper =
+        Escapers.builder()
```

Comment:
Curious, did you find any patterns for `<p>` and list (e.g. `<ol>`) escaping?

Reply:
At least in baseline file, all `<p> <ol>` components are hard-coded in monolith (for example: [settings.snip](https://github.com/googleapis/gapic-generator/blob/master/src/main/resources/com/google/api/codegen/java/settings.snip#L30) I iterated all of them, and can find corresponding template in monolith generator) so I think we do not have escape them here. But I do find in `java-pubsub` that some `<p>` which are neither in the monolith template or proto files. I suspect it comes from samplegen, but double check that.